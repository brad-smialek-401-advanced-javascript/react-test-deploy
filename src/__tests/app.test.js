import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Counter from '../components/counter/counter.js';
import App from '../components/app.js';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('< Counter /> ' , () => {

  it('is alive at start', () => {
    let components = shallow(<Counter />);
    expect(components.find('div').exists()).toBeTruthy();
  });

  it('changes state on a click', () => {
    let components = mount(<Counter />);
    let buttonDown = components.find('#clickdown');
    let buttonUp = components.find('#clickup');

    buttonDown.simulate('click')
    expect(components.state('polarity')).toBe('negative');
    buttonUp.simulate('click');
    expect(components.state('polarity')).toBe('neutral');
    buttonUp.simulate('click');
    expect(components.state('polarity')).toBe('positive');
  });

  it('renders correctly', () => {
    const thing = renderer.create(<Counter/>).toJSON();
    expect(thing).toMatchSnapshot();
  })
});