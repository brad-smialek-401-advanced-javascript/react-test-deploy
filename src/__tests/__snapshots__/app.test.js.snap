// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Counter />  renders correctly 1`] = `
<div
  id="clickCount"
>
  <button
    className="clicker"
    id="clickdown"
    onClick={[Function]}
  >
    -1
  </button>
  <h4
    className="counter "
    id="counter"
  >
    0
  </h4>
  <button
    className="clicker"
    id="clickup"
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;
